### PR TITLE
Feat locales info options

### LINKED ISSUE #21: Intl.Locale Info API
Create Documentation for ** Intl.Locale Info API** 

- [x] Review Readme [documentation and examples](https://github.com/tc39/proposal-intl-locale-info)
- [x] Create/Update MDN Main Docs Page - [Page](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Locale)

**MDN Pages :** 
- PR https://github.com/mdn/content/pull/8985
- [x] Intl.Locale.prototype.weekInfo
- [x] Intl.Locale.prototype.timeZones 
- [x] Intl.Locale.prototype.hourCycles 
- [x] Intl.Locale.prototype.numberingSystems
- [x] Intl.Locale.prototype.textInfo 
- PR https://github.com/mdn/content/pull/8768
 - [x] Intl.Locale.prototype.calendars 
 - [x]  Intl.Locale.prototype.collations

 **Browser compat-data :**  PR https://github.com/mdn/browser-compat-data/pull/12314
- [x] Intl.Locale.prototype.weekInfo
- [x] Intl.Locale.prototype.timeZones
- [x] Intl.Locale.prototype.hourCycles
- [x] Intl.Locale.prototype.numberingSystems
- [x] Intl.Locale.prototype.calendars
- [x] Intl.Locale.prototype.textInfo

### PR DESCRIPTION
**Pull request by romulocintra**
Add Info properties to Intl.Locale

Related issue : tc39/ecma402-mdn#21

This breaks down #5197  into small parts together with https://github.com/mdn/content/pull/8768

Closes tc39/ecma402-mdn#21

### CODE REVIEW

File: files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+## Examples
+
+### Obtaining supported calendars
+
+List supported hour cycles for a given `Locale`.
+
+```js
+let arEG = new Intl.Locale("ar-EG");
+console.log(arEG.hourCycles); // Prints ["h12"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md:41]:
```suggestion
console.log(arEG.hourCycles); // logs ["h12"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+```js
+let jaJP = new Intl.Locale("ja-JP");
+console.log(jaJP.hourCycles); // Prints ["h23"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md:46]:
```suggestion
console.log(jaJP.hourCycles); // logs ["h23"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  - Locale
+  - Localization
+  - Prototype
+  - Reference
+  - hourCycles
+browser-compat: javascript.builtins.Intl.Locale.hourCycles
+---
+{{JSRef}}
+
+The **`Intl.Locale.prototype.hourCycles`** property is an accessor property which returns a list of 1 or more unique hour cycle identifiers for the `Locale`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md:18]:
```suggestion
The **`Intl.Locale.prototype.hourCycles`** property is an accessor property which returns a list of one or more unique hour cycle identifiers for the `Locale`.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+## Description
+
+There are 2 main types of time keeping conventions (clocks) used around the world: the 12 hour clock and the 24 hour clock. The `hourCycles` property makes it easier for JavaScript programmers to access all clock types available for a particular locale. Like other additional locale data, hour cycle type is an [extension subtag](https://www.unicode.org/reports/tr35/#u_Extension), which extends the data contained in a locale string. The hour cycle type can have several different values, which are listed in the table below.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/hourcycles/index.md:22]:
```suggestion
There are two main types of time keeping conventions (clocks) used around the world: the 12 hour clock and the 24 hour clock. The `hourCycles` property makes it easier to access all clock types available for a particular locale. Like other additional locale data, the hour cycle type is an [extension subtag](https://www.unicode.org/reports/tr35/#u_Extension), which extends the data contained in a locale string. The hour cycle type can have several different values, which are listed in the table below.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/index.md [hunk 5]
```diff
@@ -41,16 +41,26 @@ Traditionally, the Intl API used strings to represent locales, just as Unicode d
...
+- {{jsxref("Intl/Locale/numberingSystems", "Intl.Locale.prototype.numberingSystems")}}
+  - : Returns an {{jsxref("Array")}} of numbering system identifiers available according to the locale's rules.
 - {{jsxref("Intl/Locale/numeric", "Intl.Locale.prototype.numeric")}}
   - : Returns whether the locale has special collation handling for numeric characters.
 - {{jsxref("Intl/Locale/region", "Intl.Locale.prototype.region")}}
   - : Returns the region of the world (usually a country) associated with the locale.
 - {{jsxref("Intl/Locale/script", "Intl.Locale.prototype.script")}}
   - : Returns the script used for writing the particular language used in the locale.
+- {{jsxref("Intl/Locale/textInfo", "Intl.Locale.prototype.textInfo")}}
+  - : Returns the part indicating the ordering of characters `ltr (left-to-right)` or `rtl (right-to-left)`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/index.md:59]:
```suggestion
  - : Returns the part indicating the ordering of characters `ltr` (left-to-right) or `rtl` (right-to-left).
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+  - Locale
+  - Localization
+  - Prototype
+  - Reference
+  - numberingSystems
+browser-compat: javascript.builtins.Intl.Locale.numberingSystems
+---
+{{JSRef}}
+
+The **`Intl.Locale.prototype.numberingSystems`** property is an accessor property that returns of 1 or more unique numbering system identifiers according to the [numeral system](https://en.wikipedia.org/wiki/Numeral_system) used by the `Locale`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md:18]:
```suggestion
The **`Intl.Locale.prototype.numberingSystems`** property is an accessor property that returns one or more unique numbering system identifiers according to the [numeral system](https://en.wikipedia.org/wiki/Numeral_system) used by the `Locale`.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+| tirh     | Tirhuta digits                                                             |
+| tibt     | Tibetan digits                                                             |
+| traditio | Traditional numerals — may be algorithmic                                  |
+| vaii     | Vai digits                                                                 |
+| wara     | Warang Citi digits                                                         |
+| wcho     | Wancho digits                                                              |
+
+## Examples
+
+### Obtaining  supported `numberingSystem` values via the locale string
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md:115]:
```suggestion
### Obtaining supported `numberingSystem` values via the locale string
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+In the [Unicode locale string spec](https://www.unicode.org/reports/tr35/), the values that `numberingSystem` represents correspond to the key `nu`. `nu` is considered a locale string "extension subtag". These subtags add additional data about the locale, and are added to locale identifiers by first adding the `-u` key. To set the `numberingSystem` value via the string argument to the {{jsxref("Intl/Locale/Locale", "Locale")}} constructor, first add the `-u` extension key. Next, add the `-nu` extension key to indicate that you are adding a value for `numberingSystem`. Finally, add the `numberingSystem` value to the string.
+
+List supported number system for a given `Locale`.
+
+```js
+let arEG = new Intl.Locale("ar-EG");
+console.log(arEG.numberingSystems); // Prints ["arab"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md:123]:
```suggestion
console.log(arEG.numberingSystems); // logs ["arab"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+
+```js
+let ja = new Intl.Locale("ja");
+console.log(ja.numberingSystems); // Prints ["latn"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/numberingsystems/index.md:129]:
```suggestion
console.log(ja.numberingSystems); // logs ["latn"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  - Localization
+  - Prototype
+  - Reference
+  - textInfo
+  - text direction
+browser-compat: javascript.builtins.Intl.Locale.textInfo
+---
+{{JSRef}}
+
+The **`Intl.Locale.prototype.textInfo`** property is an accessor property which returns the ordering of characters `ltr (left-to-right)` or `rtl (right-to-left)` for the associated `Locale`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md:19]:
```suggestion
The **`Intl.Locale.prototype.textInfo`** property is an accessor property which returns the ordering of characters indicated by either `ltr` (left-to-right) or by `rtl` (right-to-left) for the associated `Locale`.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+## Description
+
+Returns the `Locale` information associated with the Locale data specified in [UTS 35's Layouts Elements](https://www.unicode.org/reports/tr35/tr35-general.html#Layout_Elements).
+
+## Examples
+
+### Obtaining the Text Info
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md:27]:
```suggestion
### Obtaining text info 
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+## Examples
+
+### Obtaining the Text Info 
+
+Return the supported text directions for a given `Locale`.
+
+```js
+let ar = new Intl.Locale("ar");
+console.log(ar.textInfo); // Prints { direction: "rtl" }
+console.log(ar.textInfo.direction); // Prints "rtl"
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md:34]:
```suggestion
console.log(ar.textInfo); // logs { direction: "rtl" }
console.log(ar.textInfo.direction); // logs "rtl"
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+```
+
+```js
+let es = new Intl.Locale("es");
+console.log(es.textInfo); // Prints { direction: "ltr" }
+console.log(es.textInfo.direction); // Prints "ltr"
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/textinfo/index.md:40]:
```suggestion
console.log(es.textInfo); // logs { direction: "ltr" }
console.log(es.textInfo.direction); // logs "ltr"
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+  - Localization
+  - Prototype
+  - Reference
+  - timeZones
+  - time zone
+browser-compat: javascript.builtins.Intl.Locale.timeZones
+---
+{{JSRef}}
+
+The **`Intl.Locale.prototype.timeZones`** property is an accessor property which returns a `List` of supported time zones for choosen `Locale`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md:19]:
```suggestion
The **`Intl.Locale.prototype.timeZones`** property is an accessor property which returns an array of supported time zones for a chosen `Locale`.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+
+## Examples
+
+### Obtaining supported time zones
+
+List supported time zones for a given `Locale`.
+
+```js
+let arEG = new Intl.Locale("ar-EG");
+console.log(arEG.timeZones); // Prints ["Africa/Cairo"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md:37]:
```suggestion
console.log(arEG.timeZones); // logs ["Africa/Cairo"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+```js
+let jaJP = new Intl.Locale("ja-JP");
+console.log(jaJP.hourCycles); // Prints ["Asia/Tokyo"]
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md:42]:
```suggestion
console.log(jaJP.hourCycles); // logs ["Asia/Tokyo"]
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+```
+
+```js
+let ar = new Intl.Locale("ar");
+console.log(ar.hourCycles); // Prints undefined
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/timezones/index.md:47]:
```suggestion
console.log(ar.hourCycles); // logs undefined
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+  - Locale
+  - Localization
+  - Prototype
+  - Reference
+  - weekInfo
+browser-compat: javascript.builtins.Intl.Locale.weekInfo
+---
+{{JSRef}}
+
+The **`Intl.Locale.prototype.weekInfo`** property is an accessor property which returns the `WeekInfo` record fields `FirstDay`, `Weekend` and `MinimalDays` for the associated `Locale`.
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md:18]:
```suggestion
The **`Intl.Locale.prototype.weekInfo`** property is an accessor property which returns a `weekInfo` object with the properties `firstDay`, `weekend` and `minimalDays` for the associated `Locale`.
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+
+## Examples
+
+### Obtaining the Week Information
+
+Return the week information for a given `Locale`.
+
+```js
+let he = new Intl.Locale("he");
+console.log(he.weekInfo); // Prints {firstDay: 7, weekend: [5, 6], minimalDays: 1}
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md:33]:
```suggestion
console.log(he.weekInfo); // logs {firstDay: 7, weekend: [5, 6], minimalDays: 1}
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+
+let af = new Intl.Locale("af");
+console.log(af.weekInfo); // Prints  {firstDay: 7, weekend: [6, 7], minimalDays: 1}
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md:36]:
```suggestion
console.log(af.weekInfo); // logs  {firstDay: 7, weekend: [6, 7], minimalDays: 1}
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md [hunk 21]
```diff
@@ -0,0 +1,57 @@
...
+
+let enGB = new Intl.Locale("en-GB");
+console.log(enGB.weekInfo) // Prints  {firstDay: 1, weekend: [6, 7], minimalDays: 4}
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md:39]:
```suggestion
console.log(enGB.weekInfo) // logs  {firstDay: 1, weekend: [6, 7], minimalDays: 4}
```

File: files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md [hunk 22]
```diff
@@ -0,0 +1,57 @@
...
+
+let msBN = new Intl.Locale("en-GB");
+console.log(msBN.weekInfo) // Prints {firstDay: 7, weekend: [5, 7], minimalDays: 1}  // Brunei weekend is Friday and Sunday but not Saturday
```
Comment (Elchi3) [inline files/en-us/web/javascript/reference/global_objects/intl/locale/weekinfo/index.md:42]:
```suggestion
console.log(msBN.weekInfo) // logs {firstDay: 7, weekend: [5, 7], minimalDays: 1}  // Brunei weekend is Friday and Sunday but not Saturday 
```

**Review by Elchi3 — CHANGES_REQUESTED** [review]
Very great work! Some nits inline.

**Review by Elchi3 — APPROVED** [review]
Superb! Congrats on landing your first PR here!

**Comment by romulocintra** [comment]
> Superb! Congrats on landing your first PR here!

Thank you @Elchi3  , may I ask to have a quick l look at #8768  that is strictly related

**Pull request merged**